### DISCUSSION TITLE
Automatic opening when notification comes in the foreground

### DISCUSSION THREAD

**Discussion by Havvakksuba** [invertase/react-native-firebase#4830] [Q&A] [answered]
` messaging().onNotificationOpenedApp(remoteMessage => { Deeplink settings.. })`
Using it, the application is in the background, or if it is logged out, it redirects to the link I gave when I click the notification. But when it comes to notification in the foreground, it doesn't work. 
I used this for foreground
`messaging().onMessage (async remoteMessage => { Deeplink settings.. });`. This time, as soon as the notification arrives, it redirects to the deep link. What should I use?

**Comment by mikehardy** [comment, marked as answer]
I believe the documents on rnfirebase.io cover all of these cases.

**Reply by Havvakksuba** [reply]
I could not find which one? Usually there were problems and solutions with v5.

**Reply by mikehardy** [reply]
This to start https://rnfirebase.io/messaging/notifications - but anything more advanced (like perhaps handling notification interaction while the application is open) you must integrate a full featured device-local notifications package. There are several to choose from as noted in the docs
